applications/minotari_merge_mining_proxy/src/block_template_protocol.rs

```diff
@@ -122,87 +126,56 @@ impl BlockTemplateProtocol<'_> {
                 );
                 (data, height)
             } else {
-                let (new_template, block_template_with_coinbase, height) = match block_templates
-                    .get_new_template(best_block_hash)
-                    .await
-                {
+                let block = match self.p2pool_client.as_mut() {
+                    Some(client) => {
+                        let block_result = client.get_new_block(GetNewBlockRequest::default()).await?.into_inner();
+                        block_result
+                            .block
+                            .ok_or_else(|| MmProxyError::FailedToGetBlockTemplate("block result".to_string()))?
+                    },
                     None => {
-                        let new_template = match self.get_new_block_template().await {
-                            Ok(val) => val,
+                        let (_new_template, block_template_with_coinbase, height) = self
+                            .get_block_template_in_unnecessarily_complicated_manner(
+                                block_templates,
+                                best_block_hash,
+                                &mut loop_count,
+                            )
+                            .await?;
+
+                        match self.get_new_block(block_template_with_coinbase).await {
+                            Ok(b) => {
+                                debug!(
+                                    target: LOG_TARGET,
+                                    "Requested new block at height: #{} (try {}), block hash: `{}`",
+                                    height, loop_count,
+                                    {
+                                        let block_header = b.block.as_ref().map(|b| b.header.as_ref()).unwrap_or_default();
+                                        block_header.map(|h| h.hash.clone()).unwrap_or_default().to_hex()
+                                    },
+                                );
+                                b
+                            },
```

Comment:
We are missing handing this error
```suggestion
                            },
                    Err(MmProxyError::FailedPreconditionBlockLostRetry) => {
                        debug!(
                            target: LOG_TARGET,
                            "Chain tip has progressed past template height {}. Fetching a new block template (try {}).",
                            height, loop_count
                        );
                        continue;
                    },
```